### ISSUE TITLE
"no-useless-escape" failed with group matching `\1`

### ISSUE THREAD

**Issue by idiotWu** [eslint/eslint#7478]
**Tell us about your environment**

* **ESLint Version: 3.9.0**
* **Node Version: 6.9.0**
* **npm Version: 3.10.8**

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint.

**Please show your full configuration:**

```
"no-useless-escape": 2
```

**What did you do? Please include the actual source code causing the issue.**

```js
const regex = /require\((['"]).+?smooth-scrollbar.styl\1\);/
```

**What did you expect to happen?**

The lint should be passed.

**What actually happened? Please include the actual, raw output from ESLint.**

Lint failed with

```
12:74  error  Unnecessary escape character: \1  no-useless-escape
```

**Comment by idiotWu** [comment]
Oh I think this is a duplicate of #7472, sorry for disturbing you XD
